solutions/cses-1188.mdx

```diff
@@ -0,0 +1,127 @@
+---
+id: cses-1188
+source: CSES
+title: Bit Inversions
+author: Dustin Miao
+---
+
+We can use a Segment Tree over the given array of bits. In a node of the segment tree corresponding to range $[l, r]$, we will store three pieces of information:
+- The longest prefix of the same bit in the range $[l, r]$ (denote this as $P$)
+- The longest suffix of the same bit in the range $[l, r]$ (denote this as $S$)
+- The longest subarray of the same bit in the range $[l, r]$ (denote this as $A$)
+- The length of the subarray (equivalent to $r - l + 1$) (denote this as $L$)
```

Comment:
^ this assumes that you are looking for the longest run of a fixed bit value (e.g. the longest run of 1s)